tempodb/encoding/iterator_multiblock.go

```diff
@@ -108,32 +128,70 @@ func (i *multiblockIterator) iterate(ctx context.Context) {
 			}
 
 			comparison := bytes.Compare(currentID, lowestID)
+			foundLowerID := comparison == -1
+			shouldMerge := comparison == 0
+			// There is potential for an early dropout if comparison > 0, but it would require
+			// an additional currentMerger != nil check, since an ID will always come after
+			// an empty ID. Since checking currentMerger is already required when a lower ID
+			// is found, simply combine it with that check.
+			if currentMerger == nil || foundLowerID {
+				// currentMerger is uninitialized, or a new lowerID was found in the iterators
+				// so reset the currentMerger to use the new object. The second case shouldn't
+				// happen after merging for another ID has happened, so track it just in case.
+				metricMultiBlockIteratorWorkDropped.WithLabelValues(i.level).Add(float64(workCount))
```

Comment:
The first part of this comment just repeats the condition check using more words, so is redundant. I don't understand the second part - "The second case shouldn't happen ...". Is this code path strictly a "should never happen" warning? Or rather, if `workCount` is not 0?